src/lib/viewers/archive/ArchiveViewer.js

```diff
@@ -71,7 +71,7 @@ class ArchiveViewer extends BaseViewer {
 
         try {
             /* global BoxArchive loaded from archive.js */
-            this.archiveComponent = new BoxArchive(this.archiveEl, data);
+            this.archiveComponent = new BoxArchive(this.archiveEl, this.options.file.name, data);
```

Comment:
Just to confirm: Are `this.options.file` and `this.options.file.name` always defined?

Reply:
> Just to confirm: Are `this.options.file` and `this.options.file.name` always defined?

If file is not defined, the representations are not too, so the expected behavior is to throw the error.